acstools/utils_calib.py

```diff
@@ -464,3 +465,81 @@ def _nint(x):
     else:
         y = x - 0.5
     return int(y)
+
+
+def is_weird_subarray(filename):
+    """
+    See if the given WFC RAW file is a custom subarray CALACS cannot process.
+    Only custom subarray with overscan might be weird.
+    If that is not the case, then perhaps OSCNTAB reference file should
+    be updated instead.
+
+    Parameters
+    ----------
+    filename : str
+        The RAW file to test.
+
+    Returns
+    -------
+    is_weird : bool
+        `True` means pipeline should not process this, but should instead
+        but manually processed and then hand-deliver to pipeline.
+
+    Examples
+    --------
+    >>> import glob
+    >>> from acstools.utils_calib import is_weird_subarray
+    >>> filenames = glob.glob('*raw.fits')
+    >>> for f in filenames:
+    ...     print(f, is_weird_subarray(f))
+    j8dm03fxq_raw.fits True
+    j8eua7qqq_raw.fits False
+
+    """
+    is_weird = False
+
+    # Dimension of fullframe dark on each chip.
+    max_y = 2048
+    max_x = max_y * 2
+
+    with fits.open(filename) as pf:
+        prihdr = pf[0].header
+
+        # We won't use OSCNTAB, so don't worry about it.
+        if (prihdr['DETECTOR'] not in ('WFC', 'HRC') or
+                prihdr['BLEVCORR'] != 'PERFORM'):
+            return is_weird
+
+        scihdr = pf[1].header
+        nx = scihdr['NAXIS1']
+        ny = scihdr['NAXIS2']
+        darkfile = prihdr['DARKFILE'].replace('jref$', os.environ['jref'])
+
+        with fits.open(darkfile) as pfdrk:
+            same_size, rx, ry, x0, y0 = find_line(pf[1], pfdrk[1])
+
+        x1 = x0 + nx
+        y1 = y0 + ny
+
+        # Assume rx and ry are always 1.
```

Comment:
Add better comments here so it is clear that images which apparently do not fulfill the if statement criteria (as I recall from Tyler's testing) will (or are expected to) process correctly with CALACS.

Reply:
This is just a one-off tool. If we want to document why `rx` and `ry` are always 1, they should be done both here and in HSTCAL. I am also not sure of the best wording, so I'll have to dig around. Thanks for pointing this out.